CMakeLists.txt

```diff
@@ -66,76 +76,100 @@ install(DIRECTORY include/${PROJECT_NAME} DESTINATION include)
+                ${catkin_LIBRARIES}
+        )
+
+        target_include_directories(${PROJECT_NAME}_node
+            PRIVATE
+                ${catkin_INCLUDE_DIRS}
+        )
+
+        install(TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_node
+            ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+            LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+            RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+        )
+
+        install(DIRECTORY include/${PROJECT_NAME}/
+            DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+        )
+
+        install(DIRECTORY
+            config
+            launch
+            DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+        )
+
+        if(CATKIN_ENABLE_TESTING)
+            ## Add gtest based cpp test target and link libraries
+            catkin_add_gtest(steering_functions_test test/steering_functions_test.cpp)
+            if(TARGET steering_functions_test)
+                target_link_libraries(steering_functions_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(steering_functions_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
+
+            catkin_add_gtest(fresnel_test test/fresnel_test.cpp)
+            if(TARGET fresnel_test)
+                target_link_libraries(fresnel_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(fresnel_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
+
+            catkin_add_gtest(timing_test test/timing_test.cpp)
+            if(TARGET timing_test)
+                target_link_libraries(timing_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(timing_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
+
+            catkin_add_gtest(jacobian_test test/jacobian_test.cpp)
+            if(TARGET jacobian_test)
+                target_link_libraries(jacobian_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(jacobian_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
+
+            catkin_add_gtest(hc_cc_circle_test test/hc_cc_circle_test.cpp)
+            if(TARGET hc_cc_circle_test)
+                target_link_libraries(hc_cc_circle_test ${catkin_LIBRARIES} ${PROJECT_NAME})
+                target_include_directories(hc_cc_circle_test PRIVATE ${catkin_INCLUDE_DIRS})
+            endif()
         endif()
+    elseif(${ROS_VERSION} EQUAL 2)
+        if(BUILD_TESTING)
+            find_package(ament_cmake_gtest)
+            find_package(rclcpp REQUIRED)
 
-        catkin_add_gtest(fresnel_test test/fresnel_test.cpp)
-        if(TARGET fresnel_test)
-            target_link_libraries(fresnel_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-            target_include_directories(fresnel_test PRIVATE ${catkin_INCLUDE_DIRS})
-        endif()
+            ament_add_gtest(steering_functions_test test/steering_functions_test.cpp TIMEOUT 240)
```

Comment:
Can we remove the TIMEOUT?

Reply:
The default timeout for `ament_add_gtest` is 60 seconds, which is not enough for this test. To make the test succeed, I had to increase it